sdcm/cluster.py

```diff
@@ -1901,10 +1923,10 @@ def config_setup(self,
             if internode_compression:
                 scylla_yml['internode_compression'] = internode_compression
 
-            if ldap:
-                scylla_yml.update(self.get_ldap_config())
-            elif ms_ad_ldap:
+            if ldap and ms_ad_ldap:
                 scylla_yml.update(self.get_ldap_ms_ad_config())
+            elif ldap:
+                scylla_yml.update(self.get_ldap_config())
```

Comment:
if you are already doing many changes, this function could be renamed to `get_openldap_config()`